manticore/core/executor.py

```diff
@@ -148,14 +148,14 @@ def choice(self, state_ids):
         with self.locked_context() as policy_ctx:
             visited = policy_ctx.get('visited', dict())
             summaries = policy_ctx.get('summaries', dict())
-            lst = []
-            for id_, pc in summaries.items():
-                cnt = visited.get(pc, 0)
-                if id_ not in state_ids:
-                    continue
-                if cnt <= self._limit:
-                    lst.append((id_, visited.get(pc, 0)))
-            lst = sorted(lst, key=lambda x: x[1])
+        lst = []
+        for id_, pc in summaries.items():
+            cnt = visited.get(pc, 0)
+            if id_ not in state_ids:
+                continue
+            if cnt <= self._limit:
+                lst.append((id_, visited.get(pc, 0)))
+        lst = sorted(lst, key=lambda x: x[1])
```

Comment:
Since we are returning `lst[0][0]` at the end we should probably remove the list and keep track of the "first item" (based on `item[1]`).

Side note: when you have a sorted list and you want to insert into it and keep it being sorted, https://docs.python.org/3/library/bisect.html can be used.